src/test/unit_tests/test_namespace_fs.js

```diff
@@ -1305,6 +1334,261 @@ mocha.describe('namespace_fs copy object', function() {
             console.log('delete_object: copy twice to the same file name response', inspect(delete_copy_res));
         });
 
+
+        mocha.it(`copy directory object to same dir - size 100`, async function() {
+            const key = 'dir_to_copy_to_itself0/';
+            const res = await ns_tmp.upload_object({
+                bucket: upload_bkt,
+                key: key,
+                xattr: md1,
+                source_stream: buffer_utils.buffer_to_read_stream(data),
+                size: 100
+            }, dummy_object_sdk);
+            const file_path = ns_tmp_bucket_path + '/' + key;
+            let xattr = await get_xattr(file_path);
+            let read_md_res = await ns_tmp.read_object_md({
+                bucket: upload_bkt,
+                key: key
+            }, dummy_object_sdk);
+            assert.equal(stream_content_type, read_md_res.content_type);
+            assert.equal(res.etag, read_md_res.etag);
+            assert.deepStrictEqual(xattr, { ...add_user_prefix(read_md_res.xattr), [XATTR_DIR_CONTENT]: `${read_md_res.size}` });
+            assert.deepStrictEqual(md1, read_md_res.xattr);
```

Comment:
I think we should strip away the sort symbol before comparing here?

Reply:
thanks added it like we do on other tests